### DISCUSSION TITLE
experimental.appDir purpose?

### DISCUSSION THREAD

**Discussion by ssadler** [vercel/next.js#38551] [Help] [answered]
I was looking around for some way to have custom functions built and exported as part of the next.js build pipeline, for the purposes of making a CLI to run random database queries and etc.

It looks like this use case has specifically been avoided (happy to be corrected?), but I did find an undocumented config flag `experimental.appDir`.

From what I can see, if enabled this flag will scan for additional files to compile matching `app/*.page.(ts|js|etc)`, and it will compile them into `.next/server/app/`. The original code will be included, along with other exports (listed below).

Can someone hint what this feature is for? Thanks.

```js
exports.id = 154;
exports.ids = [154];
exports.modules = {
exports["default"] = AppRouter;
exports.fetchServerResponse = fetchServerResponse;
exports.LayoutSegmentsContext = exports.ParamsContext = exports.PathnameContext = exports.QueryContext = void 0;
exports.QueryContext = QueryContext;
exports.PathnameContext = PathnameContext;
exports.ParamsContext = ParamsContext;
exports.LayoutSegmentsContext = LayoutSegmentsContext;
exports["default"] = OuterLayoutRouter;
exports.InnerLayoutRouter = InnerLayoutRouter;
exports.matchSegment = void 0;
exports.matchSegment = matchSegment;
exports.reducer = reducer;
```

**Comment by timneutkens** [comment, marked as answer]
It's the work in progress flag for the new router which is part of https://nextjs.org/blog/layouts-rfc. Please don't use it yet as it's experimental and things will change/break.
